### ISSUE TITLE
No way to specify other-modules for custom Setup section

### ISSUE THREAD

**Issue by ezyang** [haskell/cabal#3984]
I don't know if this is affecting anyone, but there isn't any way to specify `other-modules` for a Setup script (e.g., if it spans multiple modules). A reasonable workaround to make sure the script does get added to the sdist tarball is to put it in extra-source-files, but if we ever switch to doing per-component recompilation avoidance, the lack of accurate dependency information on Setup files will prevent us from knowing when to rebuild a Setup script when you edit NOT the Setup.hs file itself.

I don't plan on fixing this unless someone actually runs into it.

**Comment by 23Skidoo** [comment]
There are some packages with multi-module Setup scripts on Hackage.

**Comment by ezyang** [comment]
Yes, but are they being actively developed? :) If not, they are unlikely to run into this bug.

**Comment by dcoutts** [comment]
Quite right, for completeness and change tracking this should be added. Not hard but not that high a priority.
